src/feature/control/control_getinfo.c

```diff
@@ -576,24 +622,16 @@ getinfo_helper_dir(control_connection_t *control_conn,
     smartlist_free(descs);
   } else if (!strcmpstart(question, "dir/status/")) {
     *answer = tor_strdup("");
-  } else if (!strcmp(question, "dir/status-vote/current/consensus")) { /* v3 */
-    if (we_want_to_fetch_flavor(get_options(), FLAV_NS)) {
-      const cached_dir_t *consensus = dirserv_get_consensus("ns");
-      if (consensus)
-        *answer = tor_strdup(consensus->dir);
-    }
-    if (!*answer) { /* try loading it from disk */
-      tor_mmap_t *mapped = networkstatus_map_cached_consensus("ns");
-      if (mapped) {
-        *answer = tor_memdup_nulterm(mapped->data, mapped->size);
-        tor_munmap_file(mapped);
-      }
-      if (!*answer) { /* generate an error */
-        *errmsg = "Could not open cached consensus. "
-          "Make sure FetchUselessDescriptors is set to 1.";
-        return -1;
-      }
-    }
+  } else if (-1 == getinfo_helper_current_consensus(
```

Comment:
I'm not sure if I like the interface here:
- If you see all the other checks in `getinfo_helper_dir()` you will notice that they only get in the `if` block if the GETINFO command matches, whereas in this patch we get in the block if there is an error. I think this is not the way it should be done, and we should copy the approach of the rest of the function. Also:

- I'd prefer if we explicitly checked the GETINFO string in this function like above, instead of hiding it into the helper function. I would probably go with:
  ```
  ...
  } else if (!strcmp(question, "dir/status-vote/current/consensus")) { /* ns consensus */
      get_info_helper_current_consensus("ns");
  } else if (!strcmp(question, "dir/status-vote/current/consensus-microdesc)) { /* md consensus */
      get_info_helper_current_consensus("nicrodesc");
  ...
  ```

- We don't usually use Yoda conditions in Tor.

Reply:
I understand, I got stuck thinking about how to move things out from the conditionals. I'll fix it.